libcudacxx/docs/extended_api/synchronization_primitives/barrier/barrier_arrive_tx.md

```diff
@@ -0,0 +1,82 @@
+---
+grand_parent: Extended API
+parent: Barriers
+---
+
+# `cuda::device::barrier_arrive_tx`
+
+Defined in header `<cuda/barrier>`:
+
+```cuda
+__device__
+cuda::barrier<cuda::thread_scope_block>::arrival_token
+cuda::device::barrier_arrive_tx(
+  cuda::barrier<cuda::thread_scope_block>& bar,
+  ptrdiff_t arrive_count_update,
+  ptrdiff_t transaction_count_update);
+```
+
+Arrives at a barrier in shared memory, updating both the arrival count and
+transaction count.
+
+## Preconditions
+
+* `__isShared(&bar) == true`
+* `1 <= arrive_count_update`
+* `0 <= transaction_count_update && transaction_count_update <= (1 << 20) - 1`
+
+
+## Effects
+
+* This function constructs an arrival_token object associated with the phase
+  synchronization point for the current phase. Then, decrements the expected
+  arrival count by `arrive_count_update` and increments the expected transaction
+  count by `transaction_count_update`.
+* This function executes atomically. The call to this function strongly
+  happens-before the start of the phase completion step for the current phase.
+* The behavior is undefined if `arrive_count_update` is less than or equal to 0
```

Comment:
This sentence is not needed, since all precondition violations are UB.

Reply:
Which sentence? This one? 

>The behavior is undefined if `arrive_count_update` is less than or equal to 0

Reply:
Right that one. There is already a precondition, and preconditon violation is UB.

Reply:
It has now been removed.